src/test/kotlin/com/terraformation/backend/tracking/db/observationStore/ObservationStoreSurvivalRateCalculationTest.kt

```diff
@@ -0,0 +1,445 @@
+      (1..count).map {
+        RecordedPlantsRow(
+            certaintyId = RecordedSpeciesCertainty.Known,
+            gpsCoordinates = point(lastCoord++),
+            speciesId = speciesId,
+            statusId = status,
+        )
+      }
+    }
+  }
+
+  private fun assertSurvivalRates(
+      expected: List<Map<Any, Map<SpeciesId, Number?>>>,
+      message: String,
+  ) {
+    val idFields =
+        listOf(
+            OBSERVED_PLOT_SPECIES_TOTALS.MONITORING_PLOT_ID,
+            OBSERVED_SUBZONE_SPECIES_TOTALS.PLANTING_SUBZONE_ID,
+            OBSERVED_ZONE_SPECIES_TOTALS.PLANTING_ZONE_ID,
+            OBSERVED_SITE_SPECIES_TOTALS.PLANTING_SITE_ID,
+        )
+
+    val levelNames = listOf("Plot", "Subzone", "Zone", "Site")
+
+    expected.forEachIndexed { index, expectedByIdMap ->
+      val actualByIdMap = fetchSurvivalRatesPerSpecies(idFields[index])
+
+      val allIdsMatch =
+          actualByIdMap.all { (id, actualSpeciesMap) ->
+            val expectedSpeciesMap = expectedByIdMap[id] ?: return@all false
+
+            actualSpeciesMap.all { (speciesId, actualRate) ->
+              val expectedRate = expectedSpeciesMap[speciesId]
+              when {
+                expectedRate == null -> actualRate == null
+                actualRate == null -> false
+                expectedRate is Double -> expectedRate.roundToInt() == actualRate
+                else -> expectedRate == actualRate
+              }
+            } &&
+                expectedSpeciesMap.all { (speciesId, _) -> actualSpeciesMap.containsKey(speciesId) }
+          } && expectedByIdMap.all { (id, _) -> actualByIdMap.containsKey(id) }
+
+      if (!allIdsMatch) {
+        val expectedString =
+            expectedByIdMap.entries.joinToString("\n") { (id, speciesMap) ->
+              "${levelNames[index]} $id: {${speciesMap.entries.joinToString(", ") { "SpeciesId: ${it.key} -> SurvivalRate: ${it.value?.let { value -> if (value is Double) value.roundToInt() else value }}" }}}"
+            }
+        val actualString =
+            actualByIdMap.entries.joinToString("\n") { (id, speciesMap) ->
+              "${levelNames[index]} $id: {${speciesMap.entries.joinToString(", ") { "SpeciesId: ${it.key} -> SurvivalRate: ${it.value}" }}}"
+            }
+        assertEquals(expectedString, actualString, "$message - ${levelNames[index]}")
+      }
+    }
+  }
+
+  private fun <ID> fetchSurvivalRatesPerSpecies(
+      idField: TableField<*, ID>
+  ): Map<ID, Map<SpeciesId, Int?>> {
+    val table = idField.table!!
+    val speciesIdField =
+        table.field("species_id", SQLDataType.BIGINT.asConvertedDataType(SpeciesIdConverter()))!!
+    val survivalRateField = table.field("survival_rate", Int::class.java)!!
+
+    return dslContext
+        .select(idField, speciesIdField, survivalRateField)
+        .from(table)
+        .where(speciesIdField.isNotNull)
+        .fetch()
+        .filter { it[idField] != null }
```

Comment:
Can this ever be null?

Reply:
I _think_ it won't cause any issues since I'll have to do a sum anyway, and nulls are treated as 0 in that case.